bin/named-person-data-export.js

```diff
@@ -0,0 +1,204 @@
+    .whereIn('c.status', [status]) // Filter by case status
+    .whereBetween('c.updated_at', [
+      moment(start).startOf('day').toISOString(),
+      moment(end).endOf('day').toISOString()
+    ]);
+}
+
+async function mergeAndSaveCSV() {
+  try {
+    // Fetch profiles based on role filter
+    const profiles = await getProfiles();
+    console.log('Profiles from DB ASL:', profiles.length);
+
+    // Extract profile_ids for filtering activity logs
+    const profileIds = profiles.map(profile => profile.profile_id);
+
+    // Fetch activity logs based on the profile_ids
+    const queryActivityLog = await getActivityLogs(profileIds);
+    console.log('Activity Logs from DB Taskflow:', queryActivityLog.length);
+
+    // Extract case_ids from activity logs
+    const caseIds = [...new Set(queryActivityLog.map(log => log.case_id))]; // Remove duplicates
+
+    // Fetch cases based on the extracted case_ids
+    const queryCases = await getCases(caseIds);
+    console.log('Cases from DB Taskflow:', queryCases.length);
+
+    // Convert cases array to a map for quick lookup
+    const casesMap = queryCases.reduce((acc, caseData) => {
+      acc[caseData.case_id] = caseData;
+      return acc;
+    }, {});
+
+    // Prepare the final result
+    const result = [];
+
+    // Keep track of cases already added per profile
+    const profileCaseMap = new Map();
+
+    profiles.forEach(profile => {
+      // Ensure each profile only gets a case once, regardless of organization
+      if (!profileCaseMap.has(profile.profile_id)) {
+        profileCaseMap.set(profile.profile_id, new Set());
+      }
+
+      queryActivityLog.forEach(log => {
+        if (log.changed_by === profile.profile_id) {
+          const caseData = casesMap[log.case_id];
+
+          if (caseData && !profileCaseMap.get(profile.profile_id).has(log.case_id)) {
+            result.push({
+              profile_id: profile.profile_id,
+              title: profile.title,
+              first_name: profile.first_name,
+              last_name: profile.last_name,
+              email: profile.email,
+              telephone: profile.telephone,
+              admin: profile.admin,
+              roles: profile.roles,
+              status: profile.status,
+              cases: `Case ID: ${caseData.case_id}, Status: ${caseData.status}, Model: ${caseData.model}, Model Status: ${caseData.model_status}`
+            });
+
+            profileCaseMap.get(profile.profile_id).add(log.case_id); // Prevent duplicate cases for this profile
+          }
+        }
+      });
+    });
+
+    // Check the result
+    console.log('Merged Result:', result);
+
+    // Write the result to CSV
+    const csvStream = fastCsv.format({
+      headers: [
+        'profile_id', 'title', 'first_name', 'last_name', 'email', 'telephone', 'admin', 'roles', 'status', 'cases'
+      ]
+    });
+
+    const writableStream = fs.createWriteStream('merged_profiles_cases.csv');
```

Comment:
If we stream to stdout then we can invoke the command via kubectl and use `kubectl exec ... > merged_profiles_cases.csv` to stream results direct to local pc. Possibly have a filename argument and stream to stdout if not provided then we can use in both ways.